# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1517601572 18000
#      Fri Feb 02 14:59:32 2018 -0500
# Node ID b5dbcfa1e3e7419a389be2340ef8e149e2ec78d8
# Parent  1c1789fa5d89809eaf5ac725129f63d4aaf77ea7
variable editor: don't set data for elements that require sub editor

* variable-editor-model.cc (variable_editor_model::setData):
Also return early if element requires a sub editor.

diff --git a/libgui/src/variable-editor-model.cc b/libgui/src/variable-editor-model.cc
--- a/libgui/src/variable-editor-model.cc
+++ b/libgui/src/variable-editor-model.cc
@@ -660,17 +660,17 @@ variable_editor_model::data (const QMode
   return QVariant ();
 }
 
 bool
 variable_editor_model::setData (const QModelIndex& idx, const QVariant& v,
                                 int role)
 {
   if (role != Qt::EditRole || v.type () != QVariant::String
-      || ! idx.isValid ())
+      || ! idx.isValid () || requires_sub_editor (idx))
     return false;
 
   // Initially, set value to whatever the user entered.
 
   int r = idx.row ();
   int c = idx.column ();
 
   QString vstr = v.toString ();
